File: C/C-Codes/binarysearch.c
#include <stdio.h>

int binarySearch(int arr[], int left, int right, int key) {
    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == key){
            return mid;
        }
        if (arr[mid] < key){
            left = mid + 1;
        }
        else{
            right = mid - 1;
        }
    }
    return -1;
}

int main() {
    int size;

    // Prompt user for the size of the array
    printf("Enter the number of elements in the array: ");
    scanf("%d", &size);

    int arr[size];

    // Prompt user to enter the elements of the array
    printf("Enter %d sorted elements:\n", size);
    for (int i = 0; i < size; i++) {
        scanf("%d", &arr[i]);
    }

    // Prompt user for the key to search
    int key;
    printf("Enter the element to search for: ");
    scanf("%d", &key);

    // Perform binary search
    int result = binarySearch(arr, 0, size - 1, key);
    if (result == -1) {
        printf("Element is not present in the array\n");
    } else {
        printf("Element is present at index %d\n", result);
    }

    return 0;
}

File: Python/binarysearch.py
def binary_search(arr, left, right, key):
    while left <= right:
        mid = left + (right - left) // 2

        if arr[mid] == key:
            return mid
        if arr[mid] < key:
            left = mid + 1
        else:
            right = mid - 1

    return -1


def main():
    size = int(input("Enter the number of elements in the array: "))

    arr = []
    print(f"Enter {size} sorted elements:")
    for i in range(size):
        arr.append(int(input()))

    key = int(input("Enter the element to search for: "))

    # Perform binary search
    result = binary_search(arr, 0, size - 1, key)
    if result == -1:
        print("Element is not present in the array")
    else:
        print(f"Element is present at index {result}")


if __name__ == "__main__":
    main()